src/classes/CRLP_ApiService.cls

```diff
@@ -67,8 +64,86 @@ public inherited sharing class CRLP_ApiService {
+     * @param parentId
+     * @param rollupTypeName
+     * @param rollupDefs
+     * @param rollupData
+     * @return Updated Rollup Definitions
+     */
+    public List<CRLP_Rollup> executeRollups(Id parentId, String rollupTypeName, List<CRLP_Rollup> rollupDefs,
+            Map<Id, Map<SObjectType, List<SObject>>> rollupData, Boolean commitToDatabase) {
+
+        // Convert the Rollup Type into an Enum
+        CRLP_RollupProcessingOptions.RollupType rollupType;
+        try {
+            rollupType = CRLP_RollupProcessingOptions.rollupTypeByName.get((String) rollupTypeName.toLowerCase());
+        } catch (Exception ex) {
+        }
+
+        if (rollupType == null) {
+            throwParameterException(CallableApiParameters.PARAM_ROLLUP_TYPE, 'No matching RollupType');
+        }
+
+        // If there are no active rollups, just return an empty collection
+        if (!CRLP_Rollup_SVC.hasActiveRollups(rollupType)) {
+            return new List<CRLP_Rollup>();
+        }
+
+        // Ensure the Rollup Definitions are populated
+        if (rollupDefs == null || rollupDefs.size() == 0) {
+            throwParameterException(CallableApiParameters.PARAM_ROLLUP_DEFS, 'Empty Rollup Definitions Collection');
+        }
+
+        // Ensure there is data to rollup for the parent record
+        if (rollupData == null || rollupData.size() == 0) {
+            throwParameterException(CallableApiParameters.PARAM_ROLLUP_DATA, 'Empty Rollup Data Collection');
+        }
 
+        // Ensure the specified ParentId is both not null and refers to an existing record.
+        if (parentId == null) {
+            throwParameterException(CallableApiParameters.PARAM_ROLLUP_PARENTID, 'A valid Parent record Id is required');
+        }
+        String countQuery = 'SELECT COUNT() FROM ' + parentId.getSobjectType().getDescribe().getName() +
+                ' WHERE Id = :parentId LIMIT 1';
+        if (Database.countQuery(countQuery) != 1) {
+            throwParameterException(CallableApiParameters.PARAM_ROLLUP_PARENTID, 'A existing Parent record is required');
+        }
+
+        try {
+            CRLP_ApiExecuteRollups rollupApiProcess = new CRLP_ApiExecuteRollups()
+                    .withRollupType(rollupType)
+                    .withParentId(parentId)
+                    .withRollupData(rollupData)
+                    .withRollupDefs(rollupDefs)
+                    .withCommitToDatabase(commitToDatabase);
+
+            rollupDefs = rollupApiProcess.executeRollups();
+        } catch (Exception ex) {
+            throwAPIException('CRLP.ExecuteRollups', '\n' + ex.getMessage() + '\n' + ex.getStackTraceString());
+        }
+        return rollupDefs;
+    }
+
+
+    /**
+     * @description Throw a custom exception there is an error parsing the parameter
+     */
+    private void throwParameterException(String paramName, String errMessage) {
+        throw new ApiException(String.format( System.Label.CallableApiParameterError,
```

Comment:
Due to formatting and indentation inconsistency, this can be hard to read. Formatting suggestion:
```
throw new ApiException(String.format(
    System.Label.CallableApiParameterError,
    new List<String>{ paramName, errMessage } 
));
```